riposte-async-http-client2/src/main/java/com/nike/riposte/client/asynchttp/AsyncHttpClientHelper.java

```diff
@@ -0,0 +1,443 @@
+     * with subspans (i.e. when {@link #performSubSpanAroundDownstreamCalls} is true). The standard/default
+     * class that is used is {@link DefaultAsyncHttpClientHelperSpanNamingAndTaggingStrategy} - if you want to
+     * adjust something, you will probably want to start with that class as a base.
+     *
+     * @param spanNamingAndTaggingStrategy The strategy to use.
+     * @return This same instance being called, to enable fluent setup.
+     */
+    public AsyncHttpClientHelper setSpanNamingAndTaggingStrategy(
+        SpanNamingAndTaggingStrategy<RequestBuilderWrapper, Response, Span> spanNamingAndTaggingStrategy
+    ) {
+        if (spanNamingAndTaggingStrategy == null) {
+            throw new IllegalArgumentException("spanNamingAndTaggingStrategy cannot be null");
+        }
+        
+        this.spanNamingAndTaggingStrategy = spanNamingAndTaggingStrategy;
+        return this;
+    }
+
+    /**
+     * Call this before one of the {@code executeAsyncHttpRequest(...)} methods in order to get a request builder you
+     * can populate with query params, headers, body, etc. If you want to specify a custom circuit breaker (or disable
+     * circuit breaking entirely) for this call then use {@link #getRequestBuilder(String, HttpMethod, Optional,
+     * boolean)} instead. This method tells the HTTP client to use a default circuit breaker based on the host being
+     * called.
+     */
+    public RequestBuilderWrapper getRequestBuilder(String url, HttpMethod method) {
+        return getRequestBuilder(url, method, Optional.empty(), false);
+    }
+
+    /**
+     * Call this before one of the {@code executeAsyncHttpRequest(...)} methods in order to get a request builder you
+     * can populate with query params, headers, body, etc. Pass in a non-empty {@code customCircuitBreaker} argument to
+     * specify the exact circuit breaker you want to use, pass in an empty {@code customCircuitBreaker} if you want the
+     * HTTP client to use a default one based on the host being called, and pass in true for the {@code
+     * disableCircuitBreaker} argument if you want to disable circuit breaking entirely for this call.
+     */
+    public RequestBuilderWrapper getRequestBuilder(String url, HttpMethod method,
+                                                   Optional<CircuitBreaker<Response>> customCircuitBreaker,
+                                                   boolean disableCircuitBreaker) {
+        RequestBuilderWrapper wrapper = generateRequestBuilderWrapper(
+            url, method, customCircuitBreaker, disableCircuitBreaker
+        );
+
+        // By default, The AsyncHttpClient doesn't properly split traffic when a DNS has multiple IP addresses associated with it,
+        //      so we have to set a name resolver that does.
+        wrapper.requestBuilder.setNameResolver(new RoundRobinInetAddressResolver(ImmediateEventExecutor.INSTANCE, new DefaultNameResolver(ImmediateEventExecutor.INSTANCE)));
```

Comment:
We want users to be able to specify their own NameResolver in case this default isn't sufficient. Maybe with the switch to a builder pattern, we can include an option for a `Function<RequestBuilderWrapper, NameResolver<InetAddress>>`?

Also, it appears that `RoundRobinInetAddressResolver` is thread-safe. We probably don't need to create a new instance for each request.

Reply:
good idea. i've added a default instance and a method on the builder to set it.